realm/src/androidTest/java/io/realm/SortTest.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import android.test.AndroidTestCase;
+
+import io.realm.entities.AllTypes;
+import io.realm.internal.TableView;
+
+public class SortTest extends AndroidTestCase {
+    private Realm testRealm = null;
+
+    private final static String FIELD_STRING = "columnString";
+    private final static String FIELD_LONG = "columnLong";
+
+    @Override
+    public void setUp() {
+        Realm.deleteRealmFile(getContext());
+        testRealm = Realm.getInstance(getContext());
+
+        testRealm.beginTransaction();
+        testRealm.clear(AllTypes.class);
+        AllTypes object1 = testRealm.createObject(AllTypes.class);
+        object1.setColumnLong(5);
+        object1.setColumnString("Adam");
+
+        AllTypes object2 = testRealm.createObject(AllTypes.class);
+        object2.setColumnLong(4);
+        object2.setColumnString("Brian");
+
+        AllTypes object3 = testRealm.createObject(AllTypes.class);
+        object3.setColumnLong(4);
+        object3.setColumnString("Adam");
+
+        AllTypes object4 = testRealm.createObject(AllTypes.class);
+        object4.setColumnLong(5);
+        object4.setColumnString("Adam");
+        testRealm.commitTransaction();
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        testRealm.close();
+    }
+
+    public void testSortMultiFailures() {
+        RealmResults<AllTypes> allTypes = testRealm.allObjects(AllTypes.class);
+
+        // zero fields specified
+        try {
+            allTypes.sort(new String[]{}, new boolean[]{});
+            fail();
+        } catch (IllegalArgumentException ignored) {}
+
+        // number of fields and sorting orders don't match
+        try {
+            allTypes.sort(new String[]{FIELD_STRING}, new boolean[]{RealmResults.SORT_ORDER_ASCENDING, RealmResults.SORT_ORDER_ASCENDING});
```

Comment:
The ascending/ascending bool array is duplicate all over. Suggest to setup and reuse once for simpler reading.